tests/TestHelpers/HttpRequestHelper.php

```diff
@@ -68,6 +70,90 @@ public static function sendRequest(
 		if ($client === null) {
 			$client = new Client();
 		}
+		$request = self::createRequest(
+			$url,
+			$method,
+			$user,
+			$password,
+			$headers,
+			$body,
+			$config,
+			$cookies,
+			$stream,
+			$timeout,
+			$client
+		);
+
+		try {
+			$response = $client->send($request);
+		} catch (BadResponseException $ex) {
+			$response = $ex->getResponse();
+
+			//if the response was null for some reason do not return it but re-throw
+			if ($response === null) {
+				throw $ex;
+			}
+		}
+		return $response;
+	}
+
+	/**
+	 * Send the requests to the server in parallel.
+	 * This function takes an array of requests and an optional client.
+	 * It will send all the requests to the server using the Pool object in guzzle.
+	 *
+	 * @param array $requests
+	 * @param Client|null $client
+	 *
+	 * @return BatchResults
+	 */
+	public static function sendBatchRequest(
+		$requests,
+		$client = null
+	) {
+		if (!isset($client)) {
```

Comment:
Only null check should be enough. `isset` checks if the variable is declared or not in addition to null-check. 

```suggestion
		if ($client !== null) {
```